src/main/java/com/bobocode/hoverla/bring/context/BeanValidator.java

```diff
@@ -1,18 +1,125 @@
 package com.bobocode.hoverla.bring.context;
 
+import com.bobocode.hoverla.bring.exception.CircularDependencyException;
+import com.bobocode.hoverla.bring.exception.IncorrectBeanNameException;
+import com.bobocode.hoverla.bring.exception.IncorrectBeanTypeException;
+import com.bobocode.hoverla.bring.exception.NoSuchBeanException;
 import com.bobocode.hoverla.bring.exception.NoUniqueBeanException;
+import lombok.extern.slf4j.Slf4j;
 
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 /**
  * Validates beans list to avoid duplicates or other conflicts
  */
+@Slf4j
 public class BeanValidator {
 
     /**
-     * @throws NoUniqueBeanException if there's bean duplicate found
+     * @param beanDefinitions all bean definitions of application context
+     * @throws IllegalArgumentException    if beanDefinitions is not initialized
+     * @throws NoUniqueBeanException       if there's bean duplicate found
```

Comment:
I don't think we need that much of custom exceptions. I suggest creating only one - `BeanValidationException`. After that we would simply pass an appropriate message to the constructor.

Reply:
done